pkg/config/registry.go

```diff
@@ -31,10 +30,9 @@ func ReadCredentials() (username, password string, err error) {
 
 // ReadCredentialsFromSecret reads from Kubernetes secret registry-credentials.
 // Returns the username and password, or error.
-func ReadCredentialsFromSecret(ctx context.Context, client client.Client) (username, password string, err error) {
+func ReadCredentialsFromSecret(ctx context.Context, client cluster.Client) (username, password string, err error) {
```

Comment:
I've reverted this change since adding the different `HelmClientFactory` implementation.

I originally changed this because I needed the type in the `HelmClientFactory` that would eventually use this to support multiple client types`client.Client` and a `kubernetes.Client`.